bulk_ghost_penalty_canvas.jl

```diff
@@ -93,24 +95,18 @@ function setup_aggregates_bounding_box_model(bgmodel, aggregate_to_cells)
                 end 
             end 
         end
-        # TO-DO: this nested loop is only valid for D=2.
-        #        Ideally, we should be able to generalize it for 
-        #        arbitrary D
-        for i in (xmin[2],xmax[2])
-            for j in (xmin[1],xmax[1])
-                bounding_box_node_coords[current]=Point(j,i)
-                current+=1
-            end 
-        end
+        bounds         = [(xmin[i], xmax[i]) for i in 1:D]
+        point_iterator = Iterators.product([bounds[i] for i in 1:D]...)
+        bounding_box_node_coords[bounding_box_node_ids[agg]] = 
+                reduce(vcat,[Point(p...) for p in point_iterator])
     end
 
-    # Set up the discrete model of bounding boxes 
-    ptr  = [ (((i-1)*2^D)+1) for i in 1:length(aggregate_to_cells)+1 ]
-    data = collect(1:length(bounding_box_node_coords))
-    bounding_box_node_ids = Gridap.Arrays.Table(data,ptr)
-
-    # TO-DO: this polytope is only valid for D=2
-    polytope=QUAD
+    # Set up the discrete model of bounding boxes
```

Comment:
Here you may use:

```
polytope = Polytope(Fill(HEX_AXIS,D)...)
```

to generalize the construction of the polytope to arbitrary space dimension. 

As we discussed, not a big deal ... we are going to run the code up to D=3 in practice.

Reply:
Cheers, this is indeed what I was looking for!